src/libfetchers/git.cc

```diff
@@ -212,9 +212,12 @@ std::pair<StorePath, Input> fetchFromWorkdir(ref<Store> store, Input & input, co
     if (fetchSettings.warnDirty)
         warn("Git tree '%s' is dirty", workdir);
 
-    auto gitOpts = Strings({ "-C", workdir, "--git-dir", gitDir, "ls-files", "-z" });
+    auto gitOpts = Strings({ "-C", workdir, "--git-dir", gitDir, "ls-files", "--cached", "-z" });
     if (submodules)
         gitOpts.emplace_back("--recurse-submodules");
+	else
+        gitOpts.emplace_back("--others");
+        gitOpts.emplace_back("--exclude-standard");
```

Comment:
Either the indentation is wrong here or there should be a `{ ... }` around the else clause.